onnx/backend/test/case/node/clip.py

```diff
@@ -16,70 +16,74 @@ class Clip(Base):
-            max=1.0
         )
 
         x = np.array([-2, 0, 2]).astype(np.float32)
-        y = np.clip(x, -1, 1)  # expected output [-1., 0., 1.]
-        expect(node, inputs=[x], outputs=[y],
+        min_val = np.array(-1).astype(np.float32)
+        max_val = np.array(1).astype(np.float32)
+        y = np.clip(x, min_val, max_val)  # expected output [-1., 0., 1.]
+        expect(node, inputs=[x, min_val, max_val], outputs=[y],
                name='test_clip_example')
 
         x = np.random.randn(3, 4, 5).astype(np.float32)
-        y = np.clip(x, -1.0, 1.0)
-        expect(node, inputs=[x], outputs=[y],
+        y = np.clip(x, min_val, max_val)
+        expect(node, inputs=[x, min_val, max_val], outputs=[y],
                name='test_clip')
         node = onnx.helper.make_node(
             'Clip',
-            inputs=['x'],
+            inputs=['x', 'min', 'max'],
             outputs=['y'],
-            min=-5.0,
-            max=5.0,
         )
 
+        min_val = np.array(-5).astype(np.float32)
+        max_val = np.array(5).astype(np.float32)
+
         x = np.array([-1, 0, 1]).astype(np.float32)
         y = np.array([-1, 0, 1]).astype(np.float32)
-        expect(node, inputs=[x], outputs=[y],
+        expect(node, inputs=[x, min_val, max_val], outputs=[y],
                name='test_clip_inbounds')
 
         x = np.array([-6, 0, 6]).astype(np.float32)
         y = np.array([-5, 0, 5]).astype(np.float32)
-        expect(node, inputs=[x], outputs=[y],
+        expect(node, inputs=[x, min_val, max_val], outputs=[y],
                name='test_clip_outbounds')
 
         x = np.array([-1, 0, 6]).astype(np.float32)
         y = np.array([-1, 0, 5]).astype(np.float32)
-        expect(node, inputs=[x], outputs=[y],
+        expect(node, inputs=[x, min_val, max_val], outputs=[y],
                name='test_clip_splitbounds')
 
     @staticmethod
     def export_clip_default():  # type: () -> None
         node = onnx.helper.make_node(
             'Clip',
-            inputs=['x'],
+            inputs=['x', 'min'],
             outputs=['y'],
-            min=0.0
         )
+        min_val = np.array(0).astype(np.float32)
         x = np.random.randn(3, 4, 5).astype(np.float32)
-        y = np.clip(x, 0.0, np.inf)
-        expect(node, inputs=[x], outputs=[y],
+        y = np.clip(x, min_val, np.inf)
+        expect(node, inputs=[x, min_val], outputs=[y],
                name='test_clip_default_min')
 
+        no_min = ""  # optional input, not supplied
         node = onnx.helper.make_node(
             'Clip',
-            inputs=['x'],
+            inputs=['x', no_min, 'max'],
             outputs=['y'],
-            max=0.0
         )
+        max_val = np.array(0).astype(np.float32)
         x = np.random.randn(3, 4, 5).astype(np.float32)
-        y = np.clip(x, -np.inf, 0.0)
-        expect(node, inputs=[x], outputs=[y],
+        y = np.clip(x, -np.inf, max_val)
+        expect(node, inputs=[x, max_val], outputs=[y],
```

Comment:
```suggestion
        expect(node, inputs=[x, "", max_val], outputs=[y],
```
maybe?

Reply:
This is correct. Please check the implementation of `expect`
https://github.com/onnx/onnx/blob/47acb06a969c3fd899f7a13270588ddb0cccd1c1/onnx/backend/test/case/node/__init__.py#L107
and another use case of optional input with operator `Scan`
https://github.com/onnx/onnx/blob/47acb06a969c3fd899f7a13270588ddb0cccd1c1/onnx/backend/test/case/node/scan.py#L43